src/milia/api/dataset.cljc

```diff
@@ -31,33 +31,35 @@
     (parse-http :get url)))
 
 #?(:clj
-    (defn- send-file-or-params
-      "Send request with file or params"
-      [method url params suppress-4xx-exceptions?]
-      (let [options (if-let [xls_file  (:xls_file params)]
-                      (multipart-options xls_file "xls_file")
-                      {:form-params params})]
-        (parse-http method url :http-options options
-                    :suppress-4xx-exceptions? suppress-4xx-exceptions?))))
+   (defn- send-file-or-params
+     "Send request with file or params"
+     [method url params suppress-4xx-exceptions?]
+     (let [options (if-let [file (:xls_file params)]
```

Comment:
I don't know how you feel about having this as a separate function and passing it params when you call it?

Reply:
makes sense, will refactor this

Reply:
Refactored to look like this https://github.com/onaio/milia/pull/294/files#diff-b2eee2b6232252cc780df1ffc11a610a2112fe32811af815f2d61667096faadbR38-R41